website/scripts/docgen/docgenParser.ts

```diff
@@ -0,0 +1,183 @@
+      prop.type.name = 'Boolean or Object';
+    }
+
+    // To deal with the platform specific props default value in ButtonGroup
+    // Input - Platform.select<typeof React.Component>({
+    //   android: TouchableNativeFeedback,
+    //   default: TouchableOpacity,
+    // })
+    // Output - TouchableOpacity(ios),TouchableNativeFeedback(android),TouchableOpacity(web)
+    if (component.name === 'ButtonGroup' && prop.name === 'Component') {
+      prop.defaultValue.value = platformMappingHandler(prop.defaultValue.value);
+    }
+
+    // To replace '\r\n' which breaks the markdown for certain props to ''
+    if (
+      prop.defaultValue &&
+      prop.defaultValue.value &&
+      prop.defaultValue.value.includes('\r\n')
+    ) {
+      prop.defaultValue.value = prop.defaultValue.value.replace(/\r\n/g, '');
+    }
+
+    // To deal with the platform specific props default value in Icon
+    // Image -  onPress ? Platform.select<typeof React.Component>({
+    //   android: TouchableNativeFeedback,
+    //   default: TouchableHighlight,
+    // }) | View
+    // Output - TouchableHighlight(ios),TouchableNativeFeedback(android),TouchableHighlight(web) | View
+    if (component.name === 'Icon' && prop.name === 'Component') {
+      prop.defaultValue.value =
+        platformMappingHandler(prop.defaultValue.value) + ' | View';
+    }
+
+    // To deal with the ViewComponent prop default value in Header
+    // linearGradientProps || !backgroundImage
+    // ? View
+    // : ImageBackground
+    // Output - ImageBackground or View Component
+    if (component.name === 'Header' && prop.name === 'ViewComponent') {
+      prop.defaultValue.value = 'ImageBackground or View Component';
+    }
+
+    // Replace all the props with default value theme?.colors?.primary to Color(Primary)
+    // Input - theme?.colors?.primary
+    // Output - Color(Primary)
+    if (
+      prop.defaultValue &&
+      prop.defaultValue.value === 'theme?.colors?.primary'
+    ) {
+      prop.defaultValue.value = 'Color(Primary)';
+    }
+
+    // To deal with the prop of default value onPress || onLongPress ? TouchableOpacity : View in Avatar
+    // Input - onPress || onLongPress ? TouchableOpacity : View
+    // Output - TouchableOpacity or View
+    if (
+      prop.defaultValue &&
+      prop.defaultValue.value ===
+        'onPress || onLongPress ? TouchableOpacity : View'
+    ) {
+      prop.defaultValue.value = 'TouchableOpacity or View';
+    }
+
+    // Filter to show the props of the components only related to the src and ignore the props of the noe modules
+    if (
+      prop.declarations !== undefined &&
+      prop.declarations.length > 0 &&
+      !componentsWithParentsTypeToBeParsed.includes(component.name)
+    ) {
+      return Boolean(
+        prop.declarations.find((declaration) => {
+          return declaration.fileName.includes(component.name);
+        })
+      );
+    }
+    return true;
+  },
```

Comment:
I am not sure whether this is maintainable in the long run.

Reply:
I know, but if we don't do this, the markdown is not parsed well with docusaurus. If we want to improve this we will have to change the code base of Typescript files such that they are parsed will with the scripts. Do you have any other solution in mind? @pranshuchittora

Reply:
I was not informed regarding this `change the code base of Typescript files such that they are parsed will with the scripts` 😞

Reply:
How much effort is required to make the following changes.

Reply:
This will take time, as we have to directly change the typescript logic such that code also works and auto generation of docs also works.

Reply:
@pranshuchittora We've already evaluated looking at how we can minimize adding custom props filter. @khushal87 has fixed certain things in the source code to minimize these but for some of these props, it's difficult / not possible to fix in the source code.

Anyway, let's relook into this separately & not block this PR. @khushal87 

We've made sure to document these things well so it's easier for maintenance.